doc/dev_guide/c_style_guide_new.rst

```diff
@@ -0,0 +1,920 @@
+
+If you have any questions about Tarantool internals, please post them on
+`StackOverflow <https://stackoverflow.com/questions/tagged/tarantool>`_ or
+ask Tarantool developers directly in `telegram <http://telegram.me/tarantool>`_.
+
+**General guidelines**
+
+The project's coding style is inspired by the `Linux kernel coding style
+<https://www.kernel.org/doc/html/v4.10/process/coding-style.html>`_.
+
+However, we have some additional guidelines, either unique to Tarantool or
+deviating from the Kernel guidelines. Below we rewrite the Linux kernel
+coding style according to the Tarantool's style features.
+
+--------------------------------------------------------------------------------
+                           Tarantool coding style
+--------------------------------------------------------------------------------
+
+This is a short document describing the preferred coding style for the
+Tarantool developers and contributors. We **insist** on following these rules
+in order to make our code consistent and understandable to any developer.
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Chapter 1: Indentation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Tabs are 8 characters, and thus indentations are
+also 8 characters. There are heretic movements that try to make indentations
+4 (or even 2!) characters deep, and that is akin to trying to define the
+value of PI to be 3.
+
+Rationale: The whole idea behind indentation is to clearly define where
+a block of control starts and ends. Especially when you've been looking
+at your screen for 20 straight hours, you'll find it a lot easier to see
+how the indentation works if you have large indentations.
+
+Now, some people will claim that having 8-character indentations makes
+the code move too far to the right, and makes it hard to read on a
+80-character terminal screen. The answer to that is that if you need
+more than 3 levels of indentation, you're screwed anyway, and should fix
+your program.
+
+In short, 8-char indents make things easier to read, and have the added
+benefit of warning you when you're nesting your functions too deep.
+Heed that warning.
+
+The preferred way to ease multiple indentation levels in a switch statement is
+to align the ``switch`` and its subordinate ``case`` labels in the same column
+instead of ``double-indenting`` the ``case`` labels. E.g.:
+
+..  code-block:: c
+
+    switch (suffix) {
+    case 'G':
+    case 'g':
+      mem <<= 30;
+      break;
+    case 'M':
+    case 'm':
+      mem <<= 20;
+      break;
+    case 'K':
+    case 'k':
+      mem <<= 10;
+      /* fall through */
+    default:
+      break;
+    }
+
+Don't put multiple statements on a single line unless you have
+something to hide:
+
+..  code-block:: c
+
+    if (condition) do_this;
+      do_something_everytime;
+
+Don't put multiple assignments on a single line either. Kernel coding style
```

Comment:
It is not Kernel coding style anymore. We must purge all mentioning of the kernel from the doc.